nautobot_circuit_maintenance/jobs/site_search.py

```diff
@@ -144,38 +124,45 @@ def run(self, data=None, commit=None):
 
         # Loop over each of the circuit maintenance records
         # pylint: disable=too-many-nested-blocks
-        counter: int = 0
+        processed_maintenance_list = []  # List of the processed maintenance records
         for circuit_maint in circuit_maintenances:
-            counter += 1
-            # Get the list of sites
-            site_list: list = []
+            # Get the set of sites
+            site_set: set = set()
             for circuit in circuit_maint.circuits:
-                site_list += get_sites_from_circuit(circuit)
+                site_set.update(get_sites_from_circuit(circuit))
 
+            processed_maintenance_list.append(circuit_maint)
+            overlapping_maintenance = False  # Flag for overlapping maintenance present
             # Check to see if there are any circuit maintenances that are overlapping at some moment in time
-            for site in site_list:
+            for site in site_set:
                 for other_circuit_maint in circuit_maintenance_mapper[site.name]:
                     # Report failures for any time where a circuit will take an outage
-                    if circuit_maint == other_circuit_maint:
+                    # Check to see if the other circuit maintenance has already had overlap checked.
+                    # If this is the same circuit maintenance record, that will also show up in the processed list
+                    # since the list is being added to before the loop.
+                    if other_circuit_maint in processed_maintenance_list:
                         continue
                     if check_for_overlap(circuit_maint, other_circuit_maint):
+                        overlapping_maintenance = True
```

Comment:
thinking on this, I believe that we should only make this `True`, when we find a dangerous overlapping. I mean, if we have the `overlap_job_exclude_no_impact` flag True, and all the impact are of this type, the overlapping is not relevant to me, so we should not mark it True, and take it into account.

Reply:
I think that I have this in place now, that it should follow along.